src/Rules/Wrappers/GhcPkg.hs

```diff
@@ -1,19 +1,35 @@
-module Rules.Wrappers.GhcPkg (ghcPkgWrapper) where
+module Rules.Wrappers.GhcPkg (inplaceGhcPkgWrapper, destGhcPkgWrapper) where
 
 import Base
 import Expression
-import Oracles.Path
+import Settings.Install
 import Settings.Path
+import Oracles.Path
 
-ghcPkgWrapper :: FilePath -> Expr String
-ghcPkgWrapper program = do
+inplaceGhcPkgWrapper :: FilePath -> FilePath -> Expr String
+inplaceGhcPkgWrapper libdir program = do
     lift $ need [sourcePath -/- "Rules/Wrappers/GhcPkg.hs"]
-    top   <- getTopDirectory
     stage <- getStage
+    top <- getTopDirectory
     -- Use the package configuration for the next stage in the wrapper.
     -- The wrapper is generated in StageN, but used in StageN+1.
     let packageDb = top -/- packageDbDirectory (succ stage)
     return $ unlines
         [ "#!/bin/bash"
-        , "exec " ++ (top -/- program)
+        , "exec " ++ (libdir -/- "bin" -/- program)
+          ++ " --global-package-db " ++ packageDb ++ " ${1+\"$@\"}" ]
+
+
+destGhcPkgWrapper :: FilePath -> FilePath -> Expr String
+destGhcPkgWrapper libdir program = do
```

Comment:
The `dest` prefix is not very clear, shouldn't it be `install`? I suggest we stick to the `inplace` vs `install` dichotomy, which seems clearer.